skyportal/handlers/api/plan.py

```diff
@@ -0,0 +1,478 @@
+    'airmass': 2.5,
+    'schedule_strategy': 'tiling',
+    'mindiff': 30.0 * 60.0,
+    'doMaxTiles': False,
+    'max_nb_tiles': 1000,
+    'doRASlice': True,
+    'raslice': [0.0, 24.0],
+}
+
+
+class PlanHandler(BaseHandler):
+    @auth_or_token
+    def post(self):
+        """
+        ---
+        description: Create an observation plan
+        tags:
+          - observingplans
+        requestBody:
+          content:
+            application/json:
+              schema: PlanHandlerPost
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        data = self.get_json()
+
+        telescope_name = data.get('telescope_name')
+        instrument_name = data.get('instrument_name')
+        dateobs = data.get('dateobs')
+        localization_name = data.get('localization_name')
+
+        telescope = (
+            Telescope.query_records_accessible_by(
+                self.current_user,
+            )
+            .filter(
+                Telescope.name == telescope_name,
+            )
+            .first()
+        )
+        if telescope is None:
+            self.error(message=f"Missing telescope {telescope_name}")
+
+        instrument = (
+            Instrument.query_records_accessible_by(
+                self.current_user,
+            )
+            .filter(
+                Instrument.telescope == telescope,
+                Instrument.name == instrument_name,
+            )
+            .first()
+        )
+        if instrument is None:
+            self.error(message=f"Missing instrument {instrument_name}")
+
+        localization = (
+            Localization.query_records_accessible_by(self.current_user)
+            .filter(
+                Localization.dateobs == dateobs,
+                Localization.localization_name == localization_name,
+            )
+            .first()
+        )
+        if localization is None:
+            self.error(message=f"Missing localization {localization_name}")
+
+        plan_args = copy.deepcopy(plan_args_default)
+
+        create_plan(self, instrument, localization, **plan_args)
+
+    @auth_or_token
+    def get(self):
+        """
+        ---
+          description: Retrieve observing plans
+          tags:
+            - observingplans
+          parameters:
+            - in: catalog_query
+              name: dateobs
+              schema:
+                type: string
+              description: Filter by event name (exact match)
+          responses:
+            200:
+              content:
+                application/json:
+                  schema: ArrayOfObservingPlan
+            400:
+              content:
+                application/json:
+                  schema: Error
+        """
+
+        data = self.get_json()
+
+        dateobs = data.get('dateobs')
```

Comment:
Error checking for when `dataobs` isn't specified

Reply:
Yes!